### ISSUE TITLE
Evaluation needs a redesign

### ISSUE THREAD

**Issue by pgonzbecer** [johnRedden/OpenGraph#83]
I almost forgot to add this, basically the current evaluation is commented out and doesn't work, because it messed up the function arithmetic. Since function arithmetic returns 2-3 when it was f(10)-f(15), we could evaluate from there, but otherwise, we need a total redesign of it

**Comment by johnRedden** [comment]
Yes I agree, function arithmetic first and then redesign evaluation based on that.  I will have a look.

y=4x
y(2) should evaluate too.

Also, f(2*3) was good to have too... we need to detect when no variable is present.

**Comment by johnRedden** [comment]
Once the function arithmetic is done... the evaluating is easy.  Just evaluate the resulting fn.  I added a property to the getUserFunction return object.

isEvaluateable: (fn(0)===NaN)?false:true

Evaluating mostly works again with a better design... still need to plot the point though... so issue not closed.

**Comment by pgonzbecer** [comment]
Design Problem! There isn't anything inherently wrong with evaluation, but function arithmetic kinda shattered how it worked before. Which point do we plot?

Say we have 3 entries
- f(x)= 2x
- y(x)=4x
- f(4) - 47 / 5 + y(y(2))

Do we go through and plot out each point of whichever is a function? If so then what about the recursive ones? Last time was simple just because it was pretty straight forward.

**Comment by johnRedden** [comment]
f(2) + g(3)  does not correspond to a point on f or g... did not think of that.  Maybe we just do not graph evaluated function arithmetic.
